weave

```diff
@@ -1197,6 +1202,9 @@ launch() {
                 IPRANGE="${1#*=}"
                 IPRANGE_SPECIFIED=1
                 ;;
+            --hairpin-mode=*)
+                HAIRPIN_MODE="${1#*=}"
```

Comment:
I think we would want it to be a general option. Hairpin mode is set by default whenever you use weave. It can be disabled via env var (when in CNI) or command-line option. This fits as well with the other options that are controlled via options to `weave` script.